### DISCUSSION TITLE
define aggregation of report diagrams on user-input (week/month/year)

### DISCUSSION THREAD

**Discussion by Bond246** [firefly-iii/firefly-iii#8841] [General]
Hello,

sometimes i'm struggeling with aggregation of report diagrams or better diagrams in general.
Mostly i want to see the evolution of metrics... so things like "what was the evolution of  income and expenditure in the last 12 months and before". Or "how was the budget evolution within the last two years" or somethink like that. It looks like that all diagrams in firefly are aggregated the same way.

But the problem i have with this is that these diagrams sometimes are aggregated in weeks, sometimes in months and timedurations over 12 months are in years. But especially this year-aggregated diagrams often are too high level to see anything else then "it is more now and was less before" without seeing any high values in a specific timerange. 
And even the weekly reports are a little confusing because... seriously... who knows weeknumbers from his mind 😉 

So how do you all deal with that? And would there be a solution to manually set the aggregation periode?

Thanks ☀️

**Comment by JC5** [comment]
That would be a great idea, and is definitely something that's on my list for the new layout. I hope to improve the reports there as well!
